### DISCUSSION TITLE
Error: unable to open database "~/Library/Messages/chat.db": authorization denied

### DISCUSSION THREAD

**Discussion by wojciech-kulik** [FelixKratz/SketchyBar#621] [Q&A] [answered]
I'm trying to run this script within sketchybar to get the number of unread Messages:
```bash
sqlite3 ~/Library/Messages/chat.db "SELECT COUNT(guid) FROM message WHERE NOT(is_read) AND NOT(is_from_me) AND text !=''"
```

However, I get this error:
```
Error: unable to open database "~/Library/Messages/chat.db": authorization denied
```

It works when I run it manually. When I print `whoami` from sketchybar it shows my username. So I'm not sure what's different. Maybe it checks also a parent process?

**Comment by 0ui** [comment, marked as answer]
I think it might require giving full disk access permission to sketchybar
https://stackoverflow.com/a/54203029
